src/Zigzag_persistence/concept/ZigzagOptions.h

```diff
@@ -0,0 +1,79 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Hannah Schreiber
+ *
+ *    Copyright (C) 2024 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+#ifndef CONCEPT_ZZ_OPTIONS_TYPE_H_
+#define CONCEPT_ZZ_OPTIONS_TYPE_H_
+
+/** @file ZigzagOptions.h
+ * @brief Contains @ref Gudhi::zigzag_persistence::ZigzagOptions and
+ * @ref Gudhi::zigzag_persistence::FilteredZigzagOptions concept.
+ */
+
+namespace Gudhi {
+namespace zigzag_persistence {
+
+/**
+ * @ingroup zigzag_persistence
+ *
+ * @brief List of options used for the filtered zigzag persistence computation.
+ */
+struct FilteredZigzagOptions {
+  /**
+   * @brief Type for the face IDs used internally and other indexations. It must be signed.
+   */
+  using internal_key = unspecified;
+
+  /**
+   * @brief Type for the face IDs used at insertion and in the boundaries given as argument.
+   */
+  using face_key = unspecified;
+
+  /**
+   * @brief Type for filtration values.
+   */
+  using filtration_value = unspecified;
```

Comment:
@VincentRouvreau would have to weigh in, but I think the type `Filtration_value` is capitalized in all the rest of Gudhi, so it looks a bit strange here.

Reply:
I wanted to use lower case for all types which are or have high probability to be a native types and upper case for classes/structs. But true, it makes the `Filtration_value` case weird. I don't have a very strong opinion about it, so I will let you both decide.

Reply:
I don't remember where I got it from, but I think it was from a comment of someone, as I didn't paid much attention to this detail before. So either it was from someone working on Gudhi, or from one of my coauthors.

Like I said, I can change that for Gudhi if you prefer. Is it only for `Filtration_value` or is there some unwritten rule about type nomenclature?

Reply:
I don't know specifically about Gudhi (although we seem to have mostly done it this way), but it is fairly common to capitalize all types, so you can tell immediately if something is a type or a variable (in a wide sense that also includes functions). You can wait for Vincent to come back, but I think I would prefer to stick to this convention. It also makes it unnecessary (although not forbidden) to end a type name in `_t` or `_type`.